.github/workflows/eval.yml

```diff
@@ -0,0 +1,53 @@
+name: Eval
+
+on: pull_request_target
```

Comment:
We can disable approval check in the github settings for our repository. The reason to not use it is the mentioned security issues and that we cannot test changes to the CI scripts this way.

Reply:
The approval checks cannot be completely turned off unfortunately, the lowest setting is "Require approval for first-time contributors who are new to GitHub". I'm not sure what the setting was at, but last time we tried `pull_request` it was [quickly changed to `pull_request_target`](https://github.com/NixOS/nixpkgs/pull/254371).

Testing is a bit more annoying, but can be done by creating a separate PR, which is also more thorough, and I've been [doing](https://github.com/Infinisil-s-Test-Organization/nixpkgs/pulls?q=is%3Apr) [that](https://github.com/tweag/nixpkgs/pulls?q=is%3Apr) for a while. And if the PR that changes the workflow is not from a fork and instead using a Nixpkgs branch, we can also test it by making a separate PR against that branch.

And finally, `pull_request` wouldn't actually work here, because it would run without Nixpkgs secrets, which we need to request reviews (and just using `permissions: pull-requests: write` doesn't work, because it also needs permission to get org membership, because you can only request reviews from people that are in the org).

I'll take care to ensure there's a clear separation between potentially-malicious user input and secrets. This is indeed tricky as [seen before](https://github.com/NixOS/nixpkgs/pull/351446), but we have some good examples to learn from. I'd also appreciate @basti564 and 13x1's review here, and ideally also a link to the slides on GitHub Actions security you showed me at NixCon :)